client/src/app/auth/auth.service.spec.ts

```diff
@@ -1,39 +1,91 @@
 import {AuthService} from './auth.service';
 import {Router} from "@angular/router";
+import {HttpClient} from "@angular/common/http";
+import {of} from "rxjs";
+import {fakeAsync, tick} from "@angular/core/testing";
 
 describe('AuthService', () => {
-  let authorizeSpy: jasmine.Spy;
-  let authenticateSpy: jasmine.Spy;
+
   let mockRouter: jasmine.SpyObj<Router>;
-  let authService: AuthService
+  let mockHttp: jasmine.SpyObj<HttpClient>;
+  let authSvc: AuthService
 
   beforeEach(() => {
     mockRouter = jasmine.createSpyObj(['navigate']);
-    authService = new AuthService(mockRouter)
-  });
+    mockHttp = jasmine.createSpyObj(['get']);
+    mockHttp.get.and.returnValue(of({}))
+    authSvc = new AuthService(mockRouter, mockHttp);
```

Comment:
I'm unfamiliar with angular.
This looks like a unit testing module is used in production code.
Is that what is happening? Is that okay?

Reply:
This is the file for the unit tests of auth.service.ts. I believe the convention is that the test file belongs in the same directory as the component/file being tested.

Reply:
yeah .spec.ts files are test("specification") files